recipes/hdfs_user.rb

```diff
@@ -0,0 +1,18 @@
+
+group node['hops']['secure_group'] do
+  action :modify
+  members node['hops']['hdfs']['user']
+  append true
+  not_if { node['install']['external_users'].casecmp("true") == 0 }
+end
+
+crypto_dir = x509_helper.get_crypto_dir(node['hops']['hdfs']['user'])
+kagent_hopsify "Generate x.509" do
+  user node['hops']['hdfs']['user']
+  crypto_directory crypto_dir
+  common_name "namenode.service.consul"
```

Comment:
This is not going to work if you have multiple namenodes as both namenodes will try to generate the same certificate and the deployment will fail.

Additionally changes on the cloud platform need to be implemented as certificates are re-generated during deployment.

I'm still of the idea that the correct way of fixing this also requires flyway to run as its own user (with impersonation privileges) and not as `hdfs` user with superupser privileges.

Reply:
If we change it to a flyingduck user with hdfs superuser privileges, we still have the problem of multiple flyingduck servers. The client needs to supply the correct FQDN, otherwise  2-way TLS fails. 
What we would have to do is create a certificate for a FQDN like 'flyingduck-1.service.consul'. 
But then, when the hsfs client wants to perform a flyingduck operation, it first sends a request to Hopsworks, it will need to tell the client which flyingduck server to connect to - 'flyingduck-1.service.consul' or 'flyingduck-2.service.consul'. Because the client has to supply the correct FQDN.

Do you really want another username for on-premises deployments?